ui/src/common/actions.ts

```diff
@@ -664,9 +674,46 @@ export const StateActions = {
         trackList.push(x);
       });
     };
+    let trackLikeSrc: TrackState | TrackGroupState = state.tracks[args.srcId];
+    let trackLikeDst: TrackState | TrackGroupState = state.tracks[args.dstId];
+    let srcParent: string | undefined;
+    let dstParent: string | undefined;
+    if (trackLikeSrc && trackLikeSrc.name) {
+      srcParent = trackLikeSrc.trackGroup;
+    } else {
+      trackLikeSrc = state.trackGroups[args.srcId];
+      srcParent = trackLikeSrc.parentGroup;
+    }
+
+    if (trackLikeDst && trackLikeDst.name) {
+      dstParent = trackLikeDst.trackGroup;
+    } else {
+      trackLikeDst = state.trackGroups[args.dstId];
+      dstParent = trackLikeDst.parentGroup;
+    }
 
-    moveWithinTrackList(state.pinnedTracks);
-    moveWithinTrackList(state.scrollingTracks);
+    if (srcParent && srcParent === dstParent) {
+      if (srcParent === SCROLLING_TRACK_GROUP) {
+        moveWithinTrackList(state.scrollingTracks);
+      } else {
+        moveWithinTrackList(state.trackGroups[srcParent].sortOrder);
+      }
+    }
+    // Currently we don't support making track groups pinned
+    if ('trackGroup' in trackLikeSrc) {
```

Comment:
Should we have an `isTrack()` function similar to `isTrackArgs()`?

Reply:
For this situation it could've been solved by using the same name for trackGroup and track parents.  Other than that I'm not sure an isTrack() would be that useful.  On the other hand something to figure out what an Id identifies could be useful, but not necessary.  It would take an Id and return either a Track Group or a Track, but I'm not sure its necessary in this PR.